invenio_rdm_records/services/pids/providers/oai_invenio.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""PID Base Provider."""
+
+import json
+
+from flask import current_app
+from flask_babelex import lazy_gettext as _
+from invenio_pidstore.errors import PIDAlreadyExists, PIDDoesNotExistError
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from sqlalchemy.orm.exc import NoResultFound
+
+from .base import BaseClient, BasePIDProvider
+
+
+class OAIInvenioClient(BaseClient):
+    """OAI Client.
+
+    Loads the value from config.
+    """
+
+    def __init__(self, name, url=None, config_key=None, **kwargs):
+        """Constructor."""
+        super().__init__(name, None, None, url=url, **kwargs)
+
+        # TODO: OAISERVER_ID_PREFIX already has oai: at the beginning.
+        # guess this should be removed and only the domain name should remain?
+        self.prefix = current_app.config.get('OAISERVER_ID_PREFIX', '')
+
+
+class OAIInvenioPIDProvider(BasePIDProvider):
+    """OAI Provider class."""
+
+    name = "invenio"
+
+    def _generate_id(self, record, **kwargs):
+        """Generates an identifier value."""
+        pid_value = record.pid.pid_value
+
+        # http://www.openarchives.org/OAI/2.0/guidelines-oai-identifier.htm
+        identifier = f"oai:{self.client.prefix}:{pid_value}"
+
+        return identifier
+
+    def __init__(
+        self,
+        client,
+        pid_type="oai",
+        default_status=PIDStatus.RESERVED,
```

Comment:
Since the PID is created on publish, starting with `PIDStatus.REGISTERED` should be fine.